From 94659f844a4ccc7e815b36753fe7ef87941d9602 Mon Sep 17 00:00:00 2001
From: Mohammad Iqbal Ahmad <mahmad@marvell.com>
Date: Fri, 6 Jan 2023 16:10:55 +0530
Subject: [PATCH 209/955] examples/l3fwd: fix for coverity scan

This patch fixes (Logically dead code) coverity issue.

Coverity issue: 381687
Fixes: 6a094e328598 ("examples/l3fwd: implement FIB lookup method")

Signed-off-by: Mohammad Iqbal Ahmad <mahmad@marvell.com>
Change-Id: I23124ce6980fe630889eee8c5ac563b347b5ee3b
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/93994
Tested-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
Reviewed-by: Ashwin Sekhar T K <asekhar@marvell.com>
---
 examples/l3fwd/l3fwd_fib.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/examples/l3fwd/l3fwd_fib.c b/examples/l3fwd/l3fwd_fib.c
index edc0dd69b9fad..18398492aeb1c 100644
--- a/examples/l3fwd/l3fwd_fib.c
+++ b/examples/l3fwd/l3fwd_fib.c
@@ -359,10 +359,10 @@ fib_event_loop(struct l3fwd_event_resources *evt_rsrc,
 				nh = (uint16_t)hopsv4[ipv4_arr_assem++];
 			else
 				nh = (uint16_t)hopsv6[ipv6_arr_assem++];
-			if (nh != FIB_DEFAULT_HOP)
-				hops[i] = nh != FIB_DEFAULT_HOP ?
-						  nh :
-						  events[i].mbuf->port;
+
+			hops[i] = nh != FIB_DEFAULT_HOP ?
+				  nh :
+				  events[i].mbuf->port;
 			process_packet(events[i].mbuf, &hops[i]);
 			events[i].mbuf->port = hops[i] != BAD_PORT ?
 						       hops[i] :
-- 
2.25.1

